Sources/SwiftDocC/Model/Rendering/RenderNodeTranslator.swift

```diff
@@ -1648,12 +1652,34 @@ public struct RenderNodeTranslator: SemanticVisitor {
             return seeAlsoSections
         } ?? .init(defaultValue: [])
         
-        node.deprecationSummaryVariants = VariantCollection<[RenderBlockContent]?>(
-            from: symbol.deprecatedSummaryVariants
-        ) { _, deprecatedSummary in
-            // If there is a deprecation summary in a documentation extension file add it to the render node
-            visitMarkupContainer(MarkupContainer(deprecatedSummary.content)) as? [RenderBlockContent]
-        } ?? .init(defaultValue: nil)
+        /// The set of traits in which the symbol is deprecated in at least one platform.
+        let traitsInWhichSymbolsIsDeprecated = documentationNode.availableVariantTraits.filter { trait in
+            guard let platforms = symbol.availabilityVariants[trait]?.availability else {
+                return false
+            }
+            
+            return platforms.contains(where: { platform in
+                platform.deprecatedVersion != nil || platform.isUnconditionallyDeprecated
+            })
+        }
+        
+        node.deprecationSummaryVariants = VariantCollection(
+            from: documentationNode.availableVariantTraits,
+            fallbackDefaultValue: nil,
+            transform: { trait in
+                if traitsInWhichSymbolsIsDeprecated.contains(trait) || traitsInWhichSymbolsIsDeprecated.isEmpty {
+                    // If the symbol is marked as deprecated in this platform, or if it's not deprecated at all,
+                    // include the deprecation summary.
```

Comment:
Would it be possible to phrase this comment more in terms of _why_ instead of _what_?. For example:
```suggestion
                    // It's possible for a symbol to only be deprecated in _some_ of its language representations
                    // In this case, only display the deprecation information for those language representations.
                    //
                    // Also, previous versions of DocC treated a page as deprecated if it had a custom `@DeprecationSummmary` description, even if the symbol wasn't deprecated. 
                    // We preserve that behavior for backwards compatibility.
                    // TODO: Warn about using `@DeprecationSummmary` for non-deprecated pages, 
                    // suggesting to use `@Available` to add deprecation information.
```

Reply:
Yes, good idea. Fixed in the latest commit